### DISCUSSION TITLE
GSoC - '25

### DISCUSSION THREAD

**Discussion by AdamPSU** [openvinotoolkit/openvino#27378] [Google Summer of Code]
Hello! I'd like to know if you have plans to continue the program in '25. I ask this for three reasons:

1. Greatly interested in the project, of course
2. Don't have the C++ down yet, but willing to learn, if the program continues
3. Would like to contribute starting now, if the program continues

I know the application to GSoC '25 is as competitive to mentorship organizations as it is for the mentored students. I also know it's pretty early in the process. Still, a response is appreciated.

Best,
Adam!

**Comment by haroon0x** [comment]
Hi Adam,

I’m excited about the opportunity to work on this project and I'm glad to hear the organization plans to apply for GSoC '25!

While I may not have extensive experience with C++ yet, I do know Python and am committed to learning C++ to contribute effectively. I’ll start by exploring the open issues and getting familiar with the codebase, and I look forward to being part of the project.

Thanks again, and please let me know if there’s anything specific you’d recommend I focus on to get started.

Best,
Haroon

**Comment by ghost-2362003** [comment]
Hello, i am greatly interested in this project and would like to continue with the same
I am already familiar with python and c++ but if I can get a heads up and word of encouragement, I would be motivated to further work on these skills and make myself better so that i can contribute to your organization.

Also, if possible can somebody tell me how to create an appropriate proposal for the 2025 program to be accepted?

**Comment by mlukasze** [comment]
@adrianboguszewski fyi ^^

**Reply by AdamPSU** [reply]
He responded to my message on Linkedin; my question has been answered. :thumbsup: 

Not sure about the others, though.

**Comment by adrianboguszewski** [comment]
Yes, we plan to join GSoC 2025. It's too early to talk about any details, as the program doesn't even have a specified timeline for 2025.
However, feel free to look at our [good first issues](https://github.com/orgs/openvinotoolkit/projects/3) to get familiar with the codebase and meet potential mentors :)

**Reply by AdamPSU** [reply]
Thanks! I've already taken an issue, having lots of fun learning about OpenVINO. You'll definitely hear more about me at a more reasonable time. Sorry for bringing this up so early, haha.

**Comment by dev-KartikSharma** [comment]
Hi, I'd like to know how to get started with contribution, i am really interested in this project already familiar with python and c/c++ but am new to open source contributions. I would love to contribute to your organisation but could you please tell me how to get started?
Hoping to hear from you soon.

**Reply by adrianboguszewski** [reply]
Hi, please read this article: https://medium.com/openvino-toolkit/how-to-contribute-to-an-ai-open-source-project-c741f48e009e. Then look at: https://github.com/orgs/openvinotoolkit/projects/3

**Comment by SAIRAGARG** [comment]
Hi everyone, I’m really interested in contributing to OvenVino for GSoC! I’m excited about the project and eager to get started. Any tips on where I can begin or open issues to focus on would be super helpful!

Looking forward to contributing and learning with you all!

**Reply by adrianboguszewski** [reply]
Hi @SAIRAGARG.
Please read this article: https://medium.com/openvino-toolkit/how-to-contribute-to-an-ai-open-source-project-c741f48e009e. 
Then look at: https://github.com/orgs/openvinotoolkit/projects/3 and start working on a prerequisite task.
All the necessary information is here: https://github.com/openvinotoolkit/openvino/wiki/Google-Summer-Of-Code

**Reply by SAIRAGARG** [reply]
Thank you! I’ll pick up a task as soon as possible and get to work on making it even better.

**Comment by MehulBhundiyaH** [comment]
Hi @adrianboguszewski sir ! 
I am also interested in openvino organization, I am new to open source. Should I also follow the path which you told to SAIRAGARG ?
My skills are - Python, C++, SQL, Machine Learning and currently learning Deep Learning...

**Reply by adrianboguszewski** [reply]
Yes, the path is the same for all contributors :)

**Comment by meghana-0211** [comment]
Hello!

Can't wait to start contributing to OpenVINO. I have worked closely with Computer Vision, Machine Learning, Deep Learning and GenAI in real time projects. More closely I have worked with pytorch, keras, tensorflow and other tools and frameworks at my previous internship projects. I was always mesmerized by the world of open source, it's selflessness towards technology and what it has to offer. Last year's OpenVINO's gsoc projects were so innovative and interesting. I wanted to ask exactly how important is C++ is in OpenVINO, I am more of an expert level python developer. Can I get started regardless? 
Thank you.

**Reply by adrianboguszewski** [reply]
Hey @meghana-0211. C++ is the core of the OpenVINO toolkit as it's written in that language. Sure, you can start looking at [good first issues](https://github.com/orgs/openvinotoolkit/projects/3). However, to join GSoC you need to go through the [official process](https://summerofcode.withgoogle.com/).

**Comment by ChathuniSoori** [comment]
> Hello! I'd like to know if you have plans to continue the program in '25. I ask this for three reasons:
> 
> 1. Greatly interested in the project, of course
> 2. Don't have the C++ down yet, but willing to learn, if the program continues
> 3. Would like to contribute starting now, if the program continues
> 
> I know the application to GSoC '25 is as competitive to mentorship organizations as it is for the mentored students. I also know it's pretty early in the process. Still, a response is appreciated.
> 
> Best, Adam!
Hi Adam,

Thanks for reaching out! While I don’t have knowledge in C++ either, my background is in Java, JavaScript, HTML, CSS, and React. Like you, I’m enthusiastic about contributing to open-source projects and would love to get involved if the program continues in 2025.

It’s great to hear about your interest and willingness to learn. If you decide to explore any Java-based or web-related contributions, feel free to connect—I’d be happy to exchange ideas or share resources.

Wishing you all the best in your journey toward GSoC!

Best,
Chathuni Sooriyaarachchi

**Comment by Hmm-1224** [comment]
Hi @adrianboguszewski,

I’m excited to see that OpenVINO is participating in GSoC 2025! I’m eager to contribute and would love to get involved early to familiarize myself with the codebase and make meaningful contributions.

I am proficient in C++ and Python and have worked on several projects involving deep learning, neural networks, and model optimization. I also have experience with tools like TensorFlow, PyTorch, Keras, Caffe, ONNX, OpenCV, and OpenVINO’s inference framework.

I came across the task related to aligning the OpenVINO ONNX Frontend with the original framework implementation of the AveragePool operator for opsets 7, 10, 11, and 19. I want to ensure that I understand the task correctly:

1.Review the ONNX documentation and identify differences between opset versions.
2.Modify the existing implementation to align with the original framework while ensuring proper coverage.
3.Register the function in ops_bridge.cpp in alphabetical order.
4.Convert test models into ONNX format and add them to the ONNX models directory.
5.Add corresponding test cases for the implementation.
6.Check for xfailed Python tests, remove corresponding lines if applicable, and verify using pytest.

I have recent experience working with ONNX models, including converting models to ONNX format and optimizing them for inference. Since this task involves working with ONNX models and ensuring proper alignment with OpenVINO’s inference framework, I believe my experience is a good match.

Could you confirm if my understanding of the task is correct, or if there are any additional details I should consider? I’d love to start contributing right away.

Looking forward to your guidance!

**Reply by adrianboguszewski** [reply]
Hi @Hmm-1224. Where did you find the project you're mentioning?

**Reply by 11happy** [reply]
I think he might be talking about the issue, 
#20553

**Reply by adrianboguszewski** [reply]
Oh, ok. So this is just good first issue - a prerequisite task to apply for GSoC. All projects for this year are listed here: https://github.com/openvinotoolkit/openvino/wiki/Google-Summer-Of-Code#project-ideas-for-2025. Please note this list hasn't been finished yet.

**Reply by 11happy** [reply]
Hi @adrianboguszewski , could you please also clarify a bit on this, that would be greatly helpful. 
 https://github.com/openvinotoolkit/openvino/discussions/28885

Thank you

**Comment by AJThePro99** [comment]
Hello @adrianboguszewski !

I'm Aadith, a sophomore Computer Science student who is passionate and curious on Artificial Intelligence.

I first discovered OpenVINO a few months ago where I was working on a project where I was trying to optimize inference for a project by converting the base **Tensorflow code into IR format** for  efficient local execution on my local NPU. Though the project was eventually scrapped due to our team not advancing in a competition, OpenVINO’s capability and usage left a strong impression on me. That’s when I decided to jump down the rabbit hole and started exploring OpenVINO in depth.

___
Right now, I’m working on Ticket #20547, and PR #28682 , but I’ve hit some issues with running ONNX tests properly. If anyone has insights or has faced similar issues, I’d really appreciate it if I could get some help regarding this, more details within #20547.
And, it has really hit my progress on the prerequisite for contributing to OpenVINO for GSoC.
___

**Looking Toward GSoC 2025**
I'm very excited about contributing to OpenVINO through GSoC and am eager to see all the various project ideas for this year!

One project that really caught my attention is the **OpenVINO Messenger AI-Assistant for an AI PC**.
 After a bit of research - I've figured out that the implementation is more tricky than challenging - but that's what makes it exciting!.

By the way, based on your experience, what are the biggest roadblocks new contributors face in OpenVINO, and how do you suggest overcoming them? Also, any tips on getting PRs reviewed faster?

Looking forward to learning from all of you and contributing more.

**Reply by adrianboguszewski** [reply]
Hey @AJThePro99 for the issues related to the ticket or PR, please post in the corresponding ticket or PR.

I suggest starting early and working closely with the mentors to discuss the final shape of the project. In case of this project it's @riacheruvu and @DimaPastushenkov

**Comment by Kaustbh** [comment]
Hi @adrianboguszewski,

My name is Kaustubh Desale, a Computer Science graduate from IIIT Kottayam, and I’m really excited about the **Gesture Control with OpenVINO project**. I have a strong background in Python and Computer Vision, with hands-on experience in OpenVINO for model optimization and deployment.

Additionally, I have experience in open-source contributions, particularly in projects like Matplotlib and fpdf2, where I have contributed to improving documentation and functionality. I am planning to apply for Google Summer of Code 2025 and would love to contribute to this project as part of the program.

Looking forward to your guidance and the opportunity to work on this!

**Reply by adrianboguszewski** [reply]
Please discuss the project with the mentors: @openvino-dev-samples and @zhuo-yoyowz

**Comment by AsVoider** [comment]
Hello @adrianboguszewski,

My name is Jianxiang Gao, and I am a senior undergraduate student majoring in Computer Science at SJTU. My research focuses on MLSYS, particularly on deploying models for inference on edge devices such as PCs and mobile phones. I frequently work with various inference frameworks, writing specialized operators and optimizing model inference pipelines to enhance performance. [PowerInfer](https://github.com/SJTU-IPADS/PowerInfer) is a project from my lab that I am currently dedicated to refining and optimizing. I am proficient in C, C++, and Python.

In fact, my first exposure to OpenVINO was last summer when I conducted performance benchmarking of transformer model operators on a laptop equipped with an Intel NPU, which aims at AIPC. This memorable experience gave me an initial understanding of the opset provided by OpenVINO. I am pleased to see how OpenVINO, a toolkit dedicated to model tuning and deployment, continues to evolve. I am excited about the possibility of large language models running faster on various edge devices with OpenVINO's help.

While browsing GSoC projects, I was shocked to discover that OpenVINO has been participating in GSoC for several consecutive years. This discovery sparked my curiosity, leading me to start a more in-depth study of OpenVINO. I am very interested in the **OpenVINO Messenger AI-Assistant for an AI PC project**, as it involves deploying LLMs on local AI PCs and integrating RAG functionality to enhance the capabilities of large models. This aligns closely with my current work. I am eager to contribute to this project in GSoC, and I would appreciate any advice and insights from the mentors to help me dive into the project more effectively.

Looking forward to the opportunity to learn from and communicate with everyone!

Lastly, here is the PR links where I solved [good first issue] [#20921](https://github.com/keras-team/keras/pull/20921) [20941](https://github.com/keras-team/keras/pull/20941)

**Reply by adrianboguszewski** [reply]
@riacheruvu  @DimaPastushenkov this is your potential contributor :)

**Comment by Simonwzm** [comment]
Hi @adrianboguszewski,

My name is Zimeng Wang, a senior undergraduate majoring in CS at SJTU. My research interests focus on MLSYS and heterogeneous computation. I'm currently involved in research projects that include enabling cooperative LLM inference across GPU and NPU, as well as designing next-generation OS-level abstractions for edge devices with multiple accelerators, such as AIPC.

Throughout my work, I've frequently studied and leveraged OpenVINO as both a powerful inference backend and a model engineering example. This has given me a foundational understanding of OpenVINO's internals and also sparked my interest in contributing to this community.

I am currently interested in **OpenVINO AI PC Model Training Kit** which fits well with my capabilities and also, I believe, is of great importance to the development of Openvino and AIPC.

I am also actively exploring the **Implement XLA plugin to run OpenVINO applications on any XLA supported devices (NVIDIA GPUs, FPGA, TPUs)**. This aligns well with my current research interest, though presents an exciting challenge to my current skill level.

I plan to investigate both ideas during the initial stages and am eager for the opportunity to learn from and collaborate with the community!

Finally, I've also having a good first issue working in progress [#21756](https://github.com/openvinotoolkit/openvino/issues/21756). 

Thank you!

Best,
Zimeng Wang

**Reply by adrianboguszewski** [reply]
Implement XLA plugin to run OpenVINO applications - @rkazants @mvafin 
OpenVINO AI PC Model Training Kit - @sbasia @omeraish

**Reply by rkazants** [reply]
Hi @Simonwzm,
Nice to meet you. Please feel free to ask question regarding "Implement XLA plugin to run OpenVINO applications on any XLA supported devices (NVIDIA GPUs, FPGA, TPUs)" project. I want to remind you that complete PRs to openvino repository is mandatory requirement for selection this project.

**Reply by sbasia** [reply]
Hi @Simonwzm,
Thanks for showing interest and glad to hear about your past research experiences especially since you have already leverage Openvino as inference backend. Please feel free to ask question regarding "OpenVINO AI PC Model Training Kit" and remember this project is very flexible and open-ended, so you can create your own design at which you are comfortable at implementing.

**Comment by Rahuldrabit** [comment]
Hi 
My name is Rahul Drabit Chowdhury. I am from the CSE department at Bangladesh University of Engineering and Technology. I have worked in VLM fine tuning recently. so , I want to contribute in below project 
Fine-tuning Vision Language Models (VLMs) for Object Detection and Hierarchical Classification using the OpenVINO Ecosystem
so, I am looking forward to contributing in above project .
Best,
Rahul

**Reply by adrianboguszewski** [reply]
The mentors are @rajeshgangireddy @samet-akcay

**Comment by Hmm-1224** [comment]
Hello @adrianboguszewski,
 can you provide openvino proposal format?

**Reply by AJThePro99** [reply]
You can find everything related to GSoC right [here](https://github.com/openvinotoolkit/openvino/wiki/Google-Summer-Of-Code)

**Reply by adrianboguszewski** [reply]
Yes, exactly. Thanks @AJThePro99 for providing the link :)

**Comment by TempestBirds729804** [comment]
Hi @adrianboguszewski,

My name is Yan Zhang, a third-year undergraduate student majoring in Computer Science at Hefei University of Technology. I’ve always been fascinated by computer vision, especially tasks like object detection and segmentation. I personally also plan to apply for a master's program in the United States for the Fall 2026 intake and pursue related research.

I am familiar with Python and C++ development and have experience working with computer vision frameworks such as OpenCV and PyTorch. I’ve also worked with CLIP-based feature extraction.

The GSOC project **“Refining Zero-Shot Object Segmentation by Combining Vision Foundation Models”** immediately caught my attention. I find the idea of combining models like DINOv2 and SAM to build a more general and robust segmentation pipeline both exciting and meaningful. I’d love to explore this further and contribute to the OpenVINO ecosystem. 🙂

My first PR is #29515.

Looking forward to learning and collaborating with the community! 

Best regards,  
Yan Zhang
